src/tools/r2rdump/R2RHeader.cs

```diff
@@ -66,34 +58,40 @@ public enum ReadyToRunFlag
         /// <param name="rva">Relative virtual address of the ReadyToRun header</param>
         /// <param name="curOffset">Index in the image byte array to the start of the ReadyToRun header</param>
         /// <exception cref="BadImageFormatException">The signature must be 0x00525452</exception>
-        public R2RHeader(byte[] image, uint rva, int curOffset)
+        public R2RHeader(byte[] image, int rva, int curOffset)
         {
             RelativeVirtualAddress = rva;
-            Offset = curOffset;
+            int startOffset = curOffset;
 
             byte[] signature = new byte[sizeof(uint)];
             Array.Copy(image, curOffset, signature, 0, sizeof(uint));
             SignatureString = System.Text.Encoding.UTF8.GetString(signature);
-            Signature = (uint)GetField(image, ref curOffset, sizeof(uint));
+            Signature = (uint)R2RReader.GetInt32(image, ref curOffset);
```

Comment:
I would recommend creating UInt variations of the R2RReader helper methods as well, so you don't have to do any type casting like you're doing here.

Reply:
TODO: Replace this GetInt32 call with GetUint32. Also not a blocking comment.

Reply:
> blocking comment

We prefer to get the comments addressed before merging PRs to master, especially if they are trivial ones like this one.